cctp_primitives/src/bit_vector/compression.rs

```diff
@@ -170,11 +183,23 @@ fn bzip2_compress(bit_vector: &[u8]) -> Result<Vec<u8>, Error> {
     Ok(bzip_compressed)
 }
 
-fn bzip2_decompress(compressed_bit_vector: &[u8]) -> Result<Vec<u8>, Error> {
-    let mut uncompressed_bitvector = Vec::new();
+fn bzip2_decompress(compressed_bit_vector: &[u8], max_decompressed_size: usize) -> Result<Vec<u8>, Error> {
+    let mut uncompressed_bitvector = Vec::with_capacity(max_decompressed_size);
     let mut decompressor = BzDecoder::new(compressed_bit_vector);
-    decompressor.read_to_end(&mut uncompressed_bitvector)?;
-    
+    let mut fixed_array = [0; DECOMPRESSION_CHUNK_SIZE];
+
+    loop {
+        let result = decompressor.read(&mut fixed_array);
+        let read_size = result?;
+        uncompressed_bitvector.extend_from_slice(&fixed_array[..read_size]);
+
+        if read_size == 0 { break; }
+
+        if uncompressed_bitvector.len() > max_decompressed_size { Err(format!("Max decompressed size {} exceeded {} while processing [Bzip2]", max_decompressed_size, uncompressed_bitvector.len()))? }
+    }
+
+    if uncompressed_bitvector.len() > max_decompressed_size { Err(format!("Max decompressed size {} exceeded {} while processing [Gzip]", max_decompressed_size, uncompressed_bitvector.len()))? }
```

Comment:
check duplication.
Instead you may do the check in the line 198 before the line `if read_size == 0 { break; }`

Reply:
Fixed.